tests/test_auth.py

```diff
@@ -45,6 +48,78 @@ def test_get_user_auth_token_when_file_does_not_exist(self):
         assert response == expected
 
 
+class TestValidateUserAuthTokenByProfileWithValidToken(BaseTestClass):
+    def setup(self):
+        self.token_data = "".join(random.choice(string.ascii_lowercase + string.digits) for i in range(40))
+
+        url = "{}{}".format(API_HOST_URL, URLS.profile.value)
+        headers = {"Authorization": "Token {}".format(self.token_data)}
+        responses.add(
+            responses.POST,
+            url,
+            headers=headers,
+            status=200,
+        )
+
+        self.expected = "Success: Authentication token is successfully set.\n"
+
+    @responses.activate
+    def test_validate_user_auth_token_by_profile_when_token_is_valid(self):
+        runner = CliRunner()
+        result = runner.invoke(set_token, [self.token_data])
+        response = result.output
+        assert response == self.expected
+
+
+class TestValidateUserAuthTokenByProfileWithInvalidToken(BaseTestClass):
+    def setup(self):
+        self.token_data = "".join(random.choice(string.ascii_lowercase + string.digits) for i in range(40))
+
+        url = "{}{}".format(API_HOST_URL, URLS.profile.value)
+        headers = {"Authorization": "Token".format(self.token_data)}
+        responses.add(
+            responses.POST,
+            url,
+            headers=headers,
+            status=401,
+        )
+
+        self.expected = "Failed: Could not make a authentication with this token.\n"
+
+    @responses.activate
+    def test_validate_user_auth_token_by_profile_when_token_is_invalid(self):
+        runner = CliRunner()
+        result = runner.invoke(set_token, [self.token_data])
+        response = result.output
+        assert response == self.expected
+
+
+class TestValidateUserAuthTokenByProfileWithBrokenURL(BaseTestClass):
+    def setup(self):
+        self.token_data = "".join(random.choice(string.ascii_lowercase + string.digits) for i in range(40))
```

Comment:
@takitsuse can we create a new method which takes length of random string as parameter and returns a random string. This method can be declared out of classes so that every test class can use this. This code is redundant in many places

Reply:
Sure! Should I put that function in somewhere in `tests` folder or `evalai/utils/auth.py`?